components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1263,31 +1296,237 @@ public static String getTenantDomainOfOauthApp(OAuthAppDO oAuthAppDO) {
-            return JWSAlgorithm.HS512.getName();
-        } else if ("SHA256withEC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.ES256.getName();
-        } else if ("SHA384withEC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.ES384.getName();
-        } else if ("SHA512withEC".equals(signatureAlgorithm)) {
-            return JWSAlgorithm.ES512.getName();
-        } else if (ALGORITHM_NONE.equals(signatureAlgorithm)) {
-            return JWSAlgorithm.NONE.getName();
+    public static JWSAlgorithm mapSignatureAlgorithm(String signatureAlgorithm) throws IdentityOAuth2Exception {
+
+        if (NONE.equalsIgnoreCase(signatureAlgorithm)) {
+            return new JWSAlgorithm(JWSAlgorithm.NONE.getName());
+        } else if (SHA256_WITH_RSA.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.RS256;
+        } else if (SHA384_WITH_RSA.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.RS384;
+        } else if (SHA512_WITH_RSA.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.RS512;
+        } else if (SHA256_WITH_HMAC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.HS256;
+        } else if (SHA384_WITH_HMAC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.HS384;
+        } else if (SHA512_WITH_HMAC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.HS512;
+        } else if (SHA256_WITH_EC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.ES256;
+        } else if (SHA384_WITH_EC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.ES384;
+        } else if (SHA512_WITH_EC.equals(signatureAlgorithm)) {
+            return JWSAlgorithm.ES512;
         }
         log.error("Unsupported Signature Algorithm in identity.xml");
         throw new IdentityOAuth2Exception("Unsupported Signature Algorithm in identity.xml");
     }
 
+    /**
+     * This method maps signature algorithm define in identity.xml to digest algorithms to generate the at_hash
+     *
+     * @param signatureAlgorithm
+     * @return
+     * @throws IdentityOAuth2Exception
+     */
+    public static String mapDigestAlgorithm(Algorithm signatureAlgorithm) throws IdentityOAuth2Exception {
+
+        if (JWSAlgorithm.RS256.equals(signatureAlgorithm) || JWSAlgorithm.HS256.equals(signatureAlgorithm) ||
+                JWSAlgorithm.ES256.equals(signatureAlgorithm)) {
+            return SHA256;
+        } else if (JWSAlgorithm.RS384.equals(signatureAlgorithm) || JWSAlgorithm.HS384.equals(signatureAlgorithm) ||
+                JWSAlgorithm.ES384.equals(signatureAlgorithm)) {
+            return SHA384;
+        } else if (JWSAlgorithm.RS512.equals(signatureAlgorithm) || JWSAlgorithm.HS512.equals(signatureAlgorithm) ||
+                JWSAlgorithm.ES512.equals(signatureAlgorithm)) {
+            return SHA512;
+        }
+        throw new RuntimeException("Cannot map Signature Algorithm in identity.xml to hashing algorithm");
```

Comment:
Is this really run time exception? When we read the file we know whether the configured algorithm is supported or not, which is a configuration issue. Won't it be good we log a warning that the given signature algorithm cannot be mapped and fallback for a default one? Also better to do this in a 'else' block explicitly to avoid future confusions,